helsingborg-stad/app-mitt-helsingborg
# Feat/pdf uploader

## Explain the changes you’ve made

Add components for handling PDFs, very similar to the way we upload images. 

## Explain why these changes are made

We seem to need the functionality for Stickprov.

## Explain your solution

Basically, I'm doing the exact same thing for PDFs as previously done for images. 

Same overall structure of components, with a PdfItem responsible for displaying the preview and detailed view of a PDF, and a Display component that displays a list of PdfItems. An Upload component that lets the user attach PDFs, and a Viewer that lets us show uploaded pdfs in a summary. 

Under the hood there are various things that are different, since other libraries needs to be used to display and pick PDFs compared to Images. There's also some small changes to the uploadFile helper, to handle the MIME-type for pdfs correctly. 

If people think it is warranted, one could quite easily abstract this a bit, and unify the image and pdf handling, so that some of the display code can be reused. However, I felt like making it work first, and it probably becomes a little harder to understand if such a generalization is made. 

## How to test the changes?

Checkout the branch, then run yarn to install new packages. On iOS, you need to run pod install. Then rebuild the app.

In dev, in the form for testing ImageUploader, I've added the PdfUploader component on the first page, and the viewer on the second page. To test it, you need to download one or a few pdfs to your phone/emulator, and then try and add them. Then check that they show up correctly, that pressing the thumbnail opens a modal that lets you scroll and zoom. Also check that they arrive correctly in the S3, and so on. 

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [] Building the Application on a iOS device/simulator.

## Screenshots (optional)
![pdfUploader](https://user-images.githubusercontent.com/7421890/108380643-55843f80-7207-11eb-9607-cee8073aa66f.png)
